### PR TITLE
Patch to fix #870 and add the chemical structure info as a separate table

### LINKED ISSUE #870: refactor the /chemical/ identifier query to not show SMILESes and InChIs
... and put SMILESes and InChIs in a new sections called "Structural information".

### PR DESCRIPTION
**Pull request by egonw**
This solves the problem that the SMILES and InChI were messing up the layout of the identifier table (with the value column way too wide).

### CODE REVIEW

**Comment by Daniel-Mietchen** [comment]
This works for me, but there is an odd side effect that the "Recently published works on the chemical" panel does not seem to get displayed anymore (I tried multiple times). I don't see how your code could have caused that, so it might just be some caching issue. 

In any case, I'm attaching screenshots for your proposed PR (top) and the current situation (bottom), both for https://tools.wmflabs.org/scholia/chemical/Q2270 (benzene). I also checked https://tools.wmflabs.org/scholia/chemical/Q159683 (citric acid), with the same result.

Screenshot_2019-11-29 Scholia-proposed
Screenshot_2019-11-29 Scholia-current

**Comment by egonw** [comment]
@Daniel-Mietchen, I just checked, as I still had the window open of the development process, and the list was there. At the same time, I can reproduce your observation and get stack server-side stacktrace:

```
java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:206)
	at com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)
	at com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)
	at com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)
	at com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)
	at com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
	at org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:355)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:503)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.bigdata.rdf.sail.webapp.QueryServlet$SparqlQueryTask.call(QueryServlet.java:889)
	at com.bigdata.rdf.sail.webapp.QueryServlet$SparqlQueryTask.call(QueryServlet.java:695)
	at com.bigdata.rdf.task.ApiTaskForIndexManager.call(ApiTaskForIndexManager.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
Caused by: java.lang.NullPointerException
```

I think you caught the introduction of a server side bug red handed. I'll do some more diggen and then file a Phabricator ticket.

@fnielsen, may I suggest this to not block this patch, as I'm fairly sure it's unrelated.

**Comment by egonw** [comment]
Can be reproduced on WDQS: https://query.wikidata.org/#SELECT%20%3Fdate%20%3Fwork%20%3FworkLabel%20%3Ftype%20%3Ftopics%0AWITH%20%7B%0A%20%20SELECT%20DISTINCT%20%3Fwork%20WHERE%20%7B%0A%20%20%20%20%7BINCLUDE%20%25RESULTS%0A%20%20BIND%28IRI%28REPLACE%28%3Fformatterurl%2C%20%27%5C%5C%5C%5C%241%27%2C%20str%28%3Fid%29%29%29%20AS%20%3FidUrls%29.%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22%5BAUTO_LANGUAGE%5D%2Cen%22.%20%7D%0A%20%20%20%20%20%20%3Fwork%20wdt%3AP921%20%2F%20%28wdt%3AP527%2B%20%7C%20wdt%3AP1269%2B%20%7C%20%28wdt%3AP31%2a%20%2F%20wdt%3AP279%2a%29%20%29%20wd%3AQ289372%20.%0A%20%20%20%20%7D%20UNION%20%7B%0A%20%20%20%20%20%20wd%3AQ289372%20%3Fpropp%20%3Fstatement%20.%0A%20%20%20%20%20%20%3Fstatement%20a%20wikibase%3ABestRank%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20prov%3AwasDerivedFrom%2Fpr%3AP248%20%3Fwork%20.%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%20AS%20%25works%0AWITH%20%7B%0A%20%20SELECT%20%28MAX%28%3Fdates%29%20as%20%3Fdatetime%29%20%3Fwork%20%28GROUP_CONCAT%28DISTINCT%20%3Ftype_label%3B%20separator%3D%22%2C%20%22%29%20AS%20%3Ftype%29%20%28GROUP_CONCAT%28DISTINCT%20%3Ftopic_label%3B%20separator%3D%22%20%2F%2F%20%22%29%20AS%20%3Ftopics%29%20WHERE%20%7B%0A%20%20%20%20INCLUDE%20%25works%0A%20%20%20%20%3Fwork%20wdt%3AP921%20%3Ftopic%20.%20%0A%20%20%20%20OPTIONAL%20%7B%20%3Fwork%20wdt%3AP31%20%3Ftype_%20.%20%3Ftype_%20rdfs%3Alabel%20%3Ftype_label%20.%20FILTER%20%28LANG%28%3Ftype_label%29%20%3D%20%27en%27%29%20%7D%0A%20%20%20%20OPTIONAL%20%7B%20%3Fwork%20wdt%3AP577%20%3Fdates%20.%20%7D%0A%20%20%20%20%3Ftopic%20rdfs%3Alabel%20%3Ftopic_label%20.%20%20FILTER%20%28lang%28%3Ftopic_label%29%20%3D%20%27en%27%29%0A%20%20%7D%0A%20%20GROUP%20BY%20%3Fwork%0A%7D%20AS%20%25result%0AWHERE%20%7B%0A%20%20INCLUDE%20%25result%0A%20%20BIND%28xsd%3Adate%28%3Fdatetime%29%20AS%20%3Fdate%29%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%2Cda%2Cde%2Ces%2Cfr%2Cjp%2Cnl%2Cno%2Cru%2Csv%2Czh%22.%20%7D%0A%7D%0AORDER%20BY%20DESC%28%3Fdate%29%0ALIMIT%20500%0A

**Comment by egonw** [comment]
@fnielsen, this patch it ready. The reported NPE is unrelated to this patch. It's a regression in the WDQS platform that needs exploring. I made a separate issue of this: https://github.com/fnielsen/scholia/issues/994

**Comment by fnielsen** [comment]
The "{INCLUDE %RESULTS" is what is causing Daniel's problem that I can reproduce.

File: scholia/app/templates/chemical.html [hunk 1]
```diff
@@ -98,7 +132,9 @@
 SELECT ?date ?work ?workLabel ?type ?topics
 WITH {
   SELECT DISTINCT ?work WHERE {
-    {
+    {INCLUDE %RESULTS
```
Comment (fnielsen) [inline scholia/app/templates/chemical.html]: This line and the two next seems to be wrong.

Comment (egonw) [reply scholia/app/templates/chemical.html]: oh, should have spotted that :( Sorry!

**Comment by fnielsen** [comment]
The PR is now running, e.g., https://tools.wmflabs.org/scholia/chemical/Q2270

**Pull request merged**